test/gurka/test_recost.cc

```diff
@@ -1,11 +1,199 @@
 #include "gurka.h"
+#include "mjolnir/graphtilebuilder.h"
 #include "sif/recost.h"
 #include <gtest/gtest.h>
 
 using namespace valhalla;
 
 const std::unordered_map<std::string, std::string> build_config{{"mjolnir.shortcuts", "false"}};
 
+TEST(recosting, same_historical) {
+  const std::string ascii_map = R"(A--1--B-2-3-C-----G
+                                         |     |     |
+                                         |     |     8
+                                         4     5     H
+                                         |     |
+                                         |     |
+                                         D--6--E--7--F)";
+  const gurka::ways ways = {
+      {"A1B23CG8H", {{"highway", "residential"}}},
+      {"D6E7F", {{"highway", "residential"}}},
+      {"B4D", {{"highway", "trunk"}}},
+      {"C5E", {{"highway", "trunk"}}},
+  };
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, 10);
+  std::string tile_dir = "test/data/gurka_recost";
+  auto map = gurka::buildtiles(layout, ways, {}, {}, tile_dir, build_config);
+  auto reader = std::make_shared<baldr::GraphReader>(map.config.get_child("mjolnir"));
+
+  // add historical traffic so that we can get different costs at different times
+  // TODO: move this into test utils with a callback per edge to get speed from the test
+  for (const auto& tile_id : reader->GetTileSet()) {
+    valhalla::mjolnir::GraphTileBuilder tile(tile_dir, tile_id, false);
+    std::vector<valhalla::baldr::DirectedEdge> edges;
+    edges.reserve(tile.header()->directededgecount());
+    for (const auto& edge : tile.GetDirectedEdges()) {
+      edges.push_back(edge);
+      edges.back().set_free_flow_speed(80);
+      edges.back().set_speed(55);
+      edges.back().set_constrained_flow_speed(10);
```

Comment:
here we set different types of speed values for the edges in the graph so that routing at different times and with different options will have different speed results